File: Hoja7/src/NodoArbol.java
import java.util.Comparator;

/**
 * @author Jorge Azmitia, 15202
 * @author Carlos Calderon ,15219
 * @param <T>
 * @Version 2.0
 * Clase que simula el nodo del arbol binario
 */
public class NodoArbol<T> {
	
	/* Atributos*/
	T dato;
	NodoArbol<T> i;
	NodoArbol<T> d;

	/**
	 * @param dato Es el dato de tipo T a guardar como nodo
	 */
	public NodoArbol(T dato) {
		this(dato, null, null);
	}

	/**
	 * @param dato dato contenido en el nodo
	 * @param i nodo de la izquierda
	 * @param d nodo de la derecha
	 */
	public NodoArbol(T dato, NodoArbol<T> i, NodoArbol<T> d) {
		this.i = i;
		this.d = d;
		this.dato = dato;
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see java.lang.Object#toString()
	 */
	public String toString() {
		return dato.toString();
	}
	/**
	 * Inner class para utilizar compare
	 */
	class MyComp1 implements Comparator<Integer> {
		/*
		 * (non-Javadoc)
		 * 
		 * @see java.util.Comparator#compare(java.lang.Object, java.lang.Object)
		 */
		public int compare(Integer x, Integer y) {
			return y - x;
		}
	}
}

File: src/NodoArbol.scala
import java.util.Comparator
import scala.collection.JavaConversions._

class NodoArbol[T](var dato: T, var i: NodoArbol[T], var d: NodoArbol[T]) {

var dato1: T = dato;
var i1: NodoArbol[T] = i;
var d1: NodoArbol[T] = d;

def this(dato: T) {
	this(dato, null, null)
}

override def toString(): String = dato.toString

class MyComp1 extends Comparator[Integer] {

	def compare(x: java.lang.Integer, y: java.lang.Integer): Int = y - x
}
}